src/video/SDL_video.c

```diff
@@ -2646,9 +2646,13 @@ bool SDL_RecreateWindow(SDL_Window *window, SDL_WindowFlags flags)
         SDL_Vulkan_UnloadLibrary();
     }
 
+    // Don't actually recreate the window in Emscripten because the SDL_WindowData's
+    // canvas_id and keyboard_element would be lost if destroyed
+#ifndef SDL_PLATFORM_EMSCRIPTEN
```

Comment:
I suggest a const bool at the top to control the changes you made here, e.g.
```c
#ifdef SDL_PLATFORM_EMSCRIPTEN
    // Don't actually recreate the window in Emscripten because the SDL_WindowData's
    // canvas_id and keyboard_element would be lost if destroyed
    const bool recreate_window = false;
#else
    const bool recreate_window = true;
#endif
```